File: gaia-sdk-javascript/src/graphql/request/input/PerceiveReceptionImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Input for reception perception impulse
 */
export class PerceiveReceptionImpulse {
public _typeName = "PerceiveReceptionImpulse";
    private data:Struct;

    constructor (data:Struct) {
        this.data = data;
    }
}

File: gaia-sdk-python/gaia_sdk/graphql/request/input/PerceiveReceptionImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class PerceiveReceptionImpulse():
    """
    Input for reception perception impulse
    """
    data: dict

    def __init__(self, data: dict):
        self.data = data

    def __eq__(self, other):
        if type(other) is type(self):
            return self.data == other.data
        return False

    def __repr__(self):
        return {'data': self.data}
